### PR TITLE
[Decomposition] Trunc

### PR DESCRIPTION
**Pull request by salilsdesai**
Summary:
Add Decomp for Trunc and add it to core_aten_decompositions

Differential Revision: D49042033

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

### CODE REVIEW

**Review by SherlockNoMad — APPROVED** [review]

File: torch/_decomp/decompositions.py [hunk 1]
```diff
@@ -4020,6 +4020,11 @@ def scaled_dot_product_flash_attention(
     )
 
 
+@register_decomposition([aten.trunc])
+def trunc(self: Tensor, **kwargs) -> Tensor:
+    return torch.where(self > 0, torch.floor(self), torch.ceil(self))
```
Comment (peterbell10) [inline torch/_decomp/decompositions.py:4025]: Why are we decomposing what is likely a hardware intrinsic operation into multiple operations? It seems to me that just because you can doesn't mean you should.

Comment (SS-JIA) [reply torch/_decomp/decompositions.py:4025]: @peterbell10 makes sense. We will remove this decomp and promote this operator to core, and provide similar treatment to other operators that map cleanly to hardware intrinsics.

Comment (kimishpatel) [reply torch/_decomp/decompositions.py:4025]: @peterbell10 to be sure I understand. If we were to do this decomp here it results in 1) perf loss and 2) potential numerics mismatch. Perf loss should be recoverable by pattern matching + fusion. 2 is a tough one and for that I agree with you that we should probably make this core aten op. But would you agree with me on 1?

Comment (peterbell10) [reply torch/_decomp/decompositions.py:4025]: I think we should also be asking how many consumers do you expect would want to "re-compose" this operator. I suspect since this is a hardware intrinsic, most backends would want it to be specialized but now they have to pattern matching this back into trunc first. So in the most common case I expect this to create more work, not save work.

On the other hand if you could show that most backends had a hard time implementing trunc and this would save a lot of maintenance then I could see the performance trade-off being worth it.
